src/java.base/share/classes/java/lang/reflect/code/Op.java

```diff
@@ -243,6 +247,28 @@ protected Op(String name, List<? extends Value> operands) {
         this.operands = List.copyOf(operands);
     }
 
+    /**
+     * Sets the originating source location of this operation, if unbound.
+     *
+     * @param l the location, a {@code null} value indicates the location is not specified.
+     * @throws IllegalStateException if this operation is bound
+     */
+    public final void setLocation(Location l) {
+        // @@@ Fail if location != null?
+        if (result != null && result.block.isBound()) {
+            throw new IllegalStateException();
+        }
+
+        location = l;
+    }
+
+    /**
+     * {@return the originating source location of this operation, otherwise {@code null} if not specified}
+     */
+    public final Location location() {
```

Comment:
Should this be returning `Optional`? Or, do we want to have a "no position" (like javac has)?

Reply:
I was pondering aliasing `null`, such as `static final NOLOCATION = null`.

Reply:
> I was pondering aliasing `null`, such as `static final Location NOLOCATION = null`.

Ok. As long as the javadoc captures the intended semantics, I don't have a strong preference.

Reply:
I follow up with another PR. At the moment it lacks clarity, we should lean more into `null` or `Optional` to signal absence.